packages/did/src/DidLinks/AccountLinks.chain.ts

```diff
@@ -0,0 +1,308 @@
+ */
+export function defaultSignerCallback(keyring: Keyring): LinkingSignerCallback {
+  return (payload: HexString, address: Address): Promise<HexString> =>
+    Promise.resolve(
+      u8aToHex(keyring.getPair(address).sign(payload, { withType: false }))
+    )
+}
+
+/**
+ * Builds an extrinsic to link `account` to a `did` where the fees and deposit are covered by some third account.
+ * This extrinsic must be authorized using the [[FullDid]] whose `didIdentifier` was used here.
+ * Note that in addition to the signing account and did used here, the submitting account will also be able to dissolve the link via reclaiming its deposit!
+ *
+ * @param accountAddress Address of the account to be linked.
+ * @param didIdentifier Method-specific identifier [[FullDid]] to be linked.
+ * @param signingCallback The signature generation callback that generates the account signature over the encoded (DidIdentifier, BlockNumber) tuple.
+ * @param nBlocksValid How many blocks into the future should the account-signed proof be considered valid?
+ * @returns An Extrinsic that must be did-authorized by the [[FullDid]] whose identifier was used.
+ */
+export async function authorizeLinkWithAccount(
+  accountAddress: Address,
+  didIdentifier: IDidIdentifier,
+  signingCallback: LinkingSignerCallback,
+  nBlocksValid = 10
+): Promise<Extrinsic> {
+  const { api } = await BlockchainApiConnection.getConnectionOrConnect()
+  const blockNo: BlockNumber = await api.query.system.number()
+  const validTill = blockNo.addn(nBlocksValid)
+  // Gets the current definition of BlockNumber (second tx argument) from the metadata.
+  const blockNumberType =
+    api.tx.didLookup.associateAccount.meta.args[1].type.toString()
+  // This is some magic on the polkadot types internals to get the DidIdentifier definition from the metadata.
+  // We get it from the connectedAccounts storage, which is a double map from (DidIdentifier, Account) -> Null.
+  const didIdentifierType = (
+    api.registry.lookup.getTypeDef(
+      // gets the type id of the keys on the connectedAccounts storage (which is a double map).
+      api.query.didLookup.connectedAccounts.creator.meta.type.asMap.key
+    ).sub as TypeDef[]
+  )[0].type // get the type of the first key, which is the DidIdentifier
+  const signMe = api
+    .createType(`(${didIdentifierType}, ${blockNumberType})`, [
+      didIdentifier,
+      validTill,
+    ])
+    .toHex()
+  let signature = u8aToU8a(await signingCallback(signMe, accountAddress))
+  const { crypto, isValid } = signatureVerify(signMe, signature, accountAddress)
+  if (!isValid) throw new Error('signature not valid')
+  const sigType = getMultiSignatureTypeFromKeypairType(crypto as KeypairType)
+  try {
+    // The signature may be prefixed; if it is, it would still verify after removing the first byte (prefix).
+    // In that case, we get rid of the prefix.
+    if (
+      signatureVerify(signMe, signature.subarray(1), accountAddress).isValid
```

Comment:
Wouldn't it be more effective to first remove the byte and do a verification and if it is valid, continue, without doing the second verification?
We would only do the second verification than in case the one with the removed byte failed, instead of always.

Reply:
Yeah that’s the one thing I still wanted to change, would be nicer

Reply:
I can change that around.

Reply:
https://github.com/KILTprotocol/sdk-js/pull/514/commits/e513d257e98878cc502f3620df71824e5045807e

Reply:
It is a bit tricky because a verifying signature can or cannot have the type annotation. So if we want to favor success cases over failures, this is what I was able to come up with.

Reply:
can we not do something like
```
let result;
try {
  result = verifySignature(signature.subarray(1))
  assert(result.isValid) // can also be if -> throw
  signature = signature.subarray(1)
} catch {
  result = verifySignature(signature)
  assert(result.isValid)
}
const {crypto} = result
```
This way we still only verify once if successful, and it's a lot shorter?

Reply:
Otherwise, this probably looks more elegant and concise.

Reply:
absolutely, we shouldn't let ourselves be held back by optimisations. on the issue of throwing, as far as I remember the error messages coming from verifySignature were actually quite helpful, so I wouldn't mind throwing them. But I can also understand if you want to catch and wrap those.

Reply:
Your proposal makes sense. And since we have to add the `<Bytes>...</Bytes>` padding to the default implementation, I think I will integrate your feedback.

Reply:
https://github.com/KILTprotocol/sdk-js/pull/514/commits/b4ddb81e5100d8a81e1911819d7722823e43b57d includes Raphael's feedback, while https://github.com/KILTprotocol/sdk-js/pull/514/commits/369ec599abc022c33eb520a69e6e7fb6bb8d1c5b wraps the default and test signing callback with the needed `wrapBytes` call.

Reply:
@rflechtner integrated your suggestion -> https://github.com/KILTprotocol/sdk-js/pull/514/commits/1c9a199eb8c696c9967cdfb5a8ca68f0354aeff4

It makes sense we expect the same on the blockchain, so good to have it padded before asking the callback to sign it.